### PR TITLE
update Reflection.getCallerClass

### PR DESCRIPTION
**Pull request by maryqyh**
added Reflection.getCallerClass and uncommented related methods

### CODE REVIEW

File: jre_emul/android/platform/libcore/ojluni/src/main/java/java/util/concurrent/atomic/AtomicIntegerFieldUpdater.java [hunk 1]
```diff
@@ -79,11 +80,8 @@
     @CallerSensitive
     public static <U> AtomicIntegerFieldUpdater<U> newUpdater(Class<U> tclass,
                                                               String fieldName) {
-        return new AtomicIntegerFieldUpdaterImpl<U>(tclass, fieldName, null);
-        /* J2ObjC: Call stack not available.
         return new AtomicIntegerFieldUpdaterImpl<U>
-            (tclass, fieldName, Reflection.getCallerClass()); // android-changed
-        */
+            (tclass, fieldName, Reflection.getCallerClass());
```
Comment (tomball) [inline jre_emul/android/platform/libcore/ojluni/src/main/java/java/util/concurrent/atomic/AtomicIntegerFieldUpdater.java:84]: Nit: restore the "// android-changed" comment from the libcore source.

File: jre_emul/android/platform/libcore/ojluni/src/main/java/java/util/concurrent/atomic/AtomicReferenceFieldUpdater.java [hunk 2]
```diff
@@ -101,11 +102,8 @@
     public static <U,W> AtomicReferenceFieldUpdater<U,W> newUpdater(Class<U> tclass,
                                                                     Class<W> vclass,
                                                                     String fieldName) {
-        return new AtomicReferenceFieldUpdaterImpl<U,W>(tclass, vclass, fieldName, null);
-        /* J2ObjC: Call stack not available.
         return new AtomicReferenceFieldUpdaterImpl<U,W>
-            (tclass, vclass, fieldName, Reflection.getCallerClass()); // android-changed
-        */
+            (tclass, vclass, fieldName, Reflection.getCallerClass());
```
Comment (tomball) [inline jre_emul/android/platform/libcore/ojluni/src/main/java/java/util/concurrent/atomic/AtomicReferenceFieldUpdater.java:106]: Ditto: restore libcore comment.

File: jre_emul/android/platform/libcore/ojluni/src/main/java/java/util/concurrent/atomic/AtomicIntegerFieldUpdater.java (hunk 1 above)
Comment (maryqyh) [reply jre_emul/android/platform/libcore/ojluni/src/main/java/java/util/concurrent/atomic/AtomicIntegerFieldUpdater.java:84]: The "// android-changed" comment is actually only in libcore-nougat but not libcore, so I delete it in this commit

File: jre_emul/android/platform/libcore/ojluni/src/main/java/java/util/concurrent/atomic/AtomicReferenceFieldUpdater.java (hunk 2 above)
Comment (maryqyh) [reply jre_emul/android/platform/libcore/ojluni/src/main/java/java/util/concurrent/atomic/AtomicReferenceFieldUpdater.java:106]: The "// android-changed" comment is only in libcore-nougat but not libcore

File: jre_emul/android/platform/libcore/ojluni/src/main/java/java/util/concurrent/atomic/AtomicIntegerFieldUpdater.java (hunk 1 above)
Comment (tomball) [reply jre_emul/android/platform/libcore/ojluni/src/main/java/java/util/concurrent/atomic/AtomicIntegerFieldUpdater.java:84]: Perfect, thanks for the update.

**Comment by tomball** [comment]
Please change Reflection.getCallerClass() to:
```
    public static Class<?> getCallerClass() {
        return VMStack.getStackClass2();
    }
```
Anjuli separately submitted the getStackClass2() update that uses a Throwable's stack frames like you did. However, we didn't skip the three stack frames from Throwable, so instead of `stack[2]` it should be `stack[5]`. I updated that in a separate CL which should be merged later this afternoon. With your change, these tests will now pass.

**Pull request merged**